tests/TestCase.php

```diff
@@ -27,12 +27,22 @@ class TestCase extends PHPUnitTestCase
 {
     use ProphecyTrait;
 
+    /**
+     * @var array<0|1, ?App> array of app instances with and without real redis. Each one may be initialised up to once per test.
+     */
+    private array $appInstance = [0 => null, 1 => null];
+
     /**
      * @return App
      * @throws Exception
      */
     protected function getAppInstance(bool $withRealRedis = false): App
     {
+        $memoizedInstance = $this->appInstance[(int)$withRealRedis];
```

Comment:
Having added a call to getAppInstance in the CreateTest setup there are now some duplicate calls to this function form within the same test. We have to return the same instance in that case.

At the end of each test PHPUnit throws away the test case object so the memoized instance(s) get garbaged.